### PR TITLE
fix: resolve data_link|label for archive docs

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |y|
|New feature?   |n|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Archived documents have no publication (es document) in the default environment. Resolve the label from the last know revision in the database.

## Dependencies:
- https://github.com/ems-project/EMSCommonBundle/pull/312

### CODE REVIEW

File: src/Repository/RevisionRepository.php [hunk 1]
```diff
@@ -42,6 +42,20 @@ public function findRevision(string $ouuid, string $contentTypeName, ?\DateTimeI
...
+        $qb = $this->createQueryBuilder('r');
+        $qb
+            ->andWhere($qb->expr()->eq('r.ouuid', ':ouuid'))
+            ->setMaxResults(1)
+            ->setParameters(['ouuid' => $ouuid])
+            ->orderBy('r.startTime', 'DESC');
+
+        $result = $qb->getQuery()->getResult();
+
+        return isset($result[0]) && $result[0] instanceof Revision ? $result[0] : null;
```
Comment (theus77) [inline src/Repository/RevisionRepository.php:56]:
```suggestion

        try{
                $result = $qb->getQuery()->getSingleResult();
                if (!$result instanceof Revision) {
                     throw new RuntimeException('Unexpected revision object';)
                }
                return $result;
               catch (NoResultException) {
                    return null;
                }
```

File: src/Twig/AppExtension.php [hunk 2]
```diff
@@ -1296,4 +1205,58 @@ public function getName(): string
...
+        if (false === $contentType = $this->contentTypeService->getByName($emsLink->getContentType())) {
+            return ['text' => $key];
+        }
+
+        $data = [];
+        if ($contentType->getIcon()) {
+            $out = '<i class="'.$contentType->getIcon().'"></i>&nbsp;&nbsp;';
+        } else {
+            $out = '<i class="fa fa-book"></i>&nbsp;&nbsp;';
+        }
```
Comment (theus77) [inline src/Twig/AppExtension.php:1228]:
```suggestion

        $out = \sprintf('<i class="%s"></i>&nbsp;&nbsp;', $contentType->getIcon() ?? 'fa fa-book');
        
```

Comment (Davidmattei) [reply src/Twig/AppExtension.php]: Fixed!

File: src/Repository/RevisionRepository.php (hunk 1 above)
Comment (Davidmattei) [reply src/Repository/RevisionRepository.php]: Fixed!

**Review by theus77 — APPROVED** [review]

**Pull request merged**